### PR TITLE
CodeCleanup should not try adding parens to tuple element names.

### LINKED ISSUE #16781: Error message when defining named arguments to an implicitly defined tuple
1. Create a VB.Net Console Project.
2. From project options, turn explicit option off.
3. Inside `Sub Main()`, type the following line and press **Enter** afterwards.

```vb.net
x = (first:=1, second:=2)
```

VS **one or more errors occurred** dialog pops up, and the following exception appears in `ActivityLog.xml`:

```
System.AggregateException: One or more errors occurred. ---&gt; System.AggregateException: One or more errors occurred. ---&gt; System.InvalidCastException: Unable to cast object of type "Microsoft.CodeAnalysis.VisualBasic.Syntax.InvocationExpressionSyntax" to type "Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax".
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitNameColonEquals(NameColonEqualsSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.NameColonEqualsSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitSimpleArgument(SimpleArgumentSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.SimpleArgumentSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SeparatedSyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitTupleExpression(TupleExpressionSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.TupleExpressionSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.VisitExpression(ExpressionSyntax node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitAssignmentStatement(AssignmentStatementSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.AssignmentStatementSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitMethodBlock(MethodBlockSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.MethodBlockSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitModuleBlock(ModuleBlockSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.ModuleBlockSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitCompilationUnit(CompilationUnitSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.CompilationUnitSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.VB$StateMachine_4_CleanupAsync.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.CodeCleanup.AbstractCodeCleanerService.&lt;IterateAllCodeCleanupProvidersAsync&gt;d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.CodeAnalysis.CodeCleanup.AbstractCodeCleanerService.&lt;CleanupAsync&gt;d__1.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Roslyn.Utilities.TaskExtensions.WaitAndGetResult[T](Task`1 task, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit.CommitFormatter.VB$StateMachine_3_CommitRegionAsync.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit.CommitBufferManager.CommitDirty(Boolean isExplicitFormat, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit.CommitViewManager._Lambda$__11-0(IWaitContext waitContext)
   at Microsoft.VisualStudio.LanguageServices.Implementation.Utilities.VisualStudioWaitIndicator.Wait(String title, String message, Boolean allowCancel, Boolean showProgress, Action`1 action)
   at Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit.CommitViewManager.OnLostAggregateFocus(Object sender, EventArgs e)
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.RaiseEvent(Object sender, EventHandler eventHandlers)
---&gt; (Inner Exception #0) System.AggregateException: One or more errors occurred. ---&gt; System.InvalidCastException: Unable to cast object of type "Microsoft.CodeAnalysis.VisualBasic.Syntax.InvocationExpressionSyntax" to type "Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax".
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitNameColonEquals(NameColonEqualsSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.NameColonEqualsSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitSimpleArgument(SimpleArgumentSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.SimpleArgumentSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SeparatedSyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitTupleExpression(TupleExpressionSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.TupleExpressionSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.VisitExpression(ExpressionSyntax node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitAssignmentStatement(AssignmentStatementSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.AssignmentStatementSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitMethodBlock(MethodBlockSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.MethodBlockSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitModuleBlock(ModuleBlockSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.ModuleBlockSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitCompilationUnit(CompilationUnitSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.CompilationUnitSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.VB$StateMachine_4_CleanupAsync.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.CodeCleanup.AbstractCodeCleanerService.&lt;IterateAllCodeCleanupProvidersAsync&gt;d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.CodeAnalysis.CodeCleanup.AbstractCodeCleanerService.&lt;CleanupAsync&gt;d__1.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Roslyn.Utilities.TaskExtensions.WaitAndGetResult[T](Task`1 task, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit.CommitFormatter.VB$StateMachine_3_CommitRegionAsync.MoveNext()
---&gt; (Inner Exception #0) System.InvalidCastException: Unable to cast object of type "Microsoft.CodeAnalysis.VisualBasic.Syntax.InvocationExpressionSyntax" to type "Microsoft.CodeAnalysis.VisualBasic.Syntax.IdentifierNameSyntax".
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitNameColonEquals(NameColonEqualsSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.NameColonEqualsSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitSimpleArgument(SimpleArgumentSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.SimpleArgumentSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SeparatedSyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitTupleExpression(TupleExpressionSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.TupleExpressionSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.VisitExpression(ExpressionSyntax node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitAssignmentStatement(AssignmentStatementSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.AssignmentStatementSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitMethodBlock(MethodBlockSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.MethodBlockSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitModuleBlock(ModuleBlockSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.ModuleBlockSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitListElement[TNode](TNode node)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitList[TNode](SyntaxList`1 list)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.VisitCompilationUnit(CompilationUnitSyntax node)
   at Microsoft.CodeAnalysis.VisualBasic.Syntax.CompilationUnitSyntax.Accept[TResult](VisualBasicSyntaxVisitor`1 visitor)
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.Rewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AddMissingTokensCodeCleanupProvider.AddMissingTokensRewriter.Visit(SyntaxNode node)
   at Microsoft.CodeAnalysis.CodeCleanup.Providers.AbstractTokensCodeCleanupProvider.VB$StateMachine_4_CleanupAsync.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.CodeAnalysis.CodeCleanup.AbstractCodeCleanerService.&lt;IterateAllCodeCleanupProvidersAsync&gt;d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.CodeAnalysis.CodeCleanup.AbstractCodeCleanerService.&lt;CleanupAsync&gt;d__1.MoveNext()&lt;---
&lt;---
```

### PR DESCRIPTION
**Pull request by VSadov**
Fixes  #16781

Because of the another bug, #16697, it is possible that tuple element names would bind to methods. That will require a bigger fix.

But regardless of that, the CodeCleanup should not try adding "()" to tuple names. No need to consult with semantic info (which is fairly expensive), when this will never be a right thing to do. It does not fit the syntax model and crashes with invalid cast exception.

**Customer scenario**

Customer gets pestered by failure popups when editing code around tuple literals. 
IDE invokes CodeCleanup, which throws. In my observation the errors get trapped and do not result in a crash of VS, but cause extremely annoying popup dialogs.

**Bugs this fixes:** 

https://github.com/dotnet/roslyn/issues/16781

**Workarounds, if any**

avoid tuple element names that match method names in VB

**Risk**

The risk is low. The fix blocks a codepath that has no chance of working.

**Performance impact**

Low. One extra condition to check in CodeCleanup. 

**Is this a regression from a previous update?**

New feature

**Root cause analysis:**

The bug shows up when editing VB tuple literals in a particular environment. - Element names should match a method declaration and some edits applied in IDE. 
While possible scenario, it would not be obvious to explicitly test it unless the problem is already known.

**How was the bug found?**

Internal use of the feature.

### CODE REVIEW

**Comment by VSadov** [comment]
@dotnet/roslyn-ide - please review.

**Comment by VSadov** [comment]
@dotnet/roslyn-compiler - FYI. not really a compiler fix.

File: src/Workspaces/CoreTest/CodeCleanup/AddMissingTokensTests.cs [hunk 1]
```diff
@@ -2595,6 +2595,43 @@ End Function
         }
 
         [Fact]
+        [Trait(Traits.Feature, Traits.Features.AddMissingTokens)]
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CoreTest/CodeCleanup/AddMissingTokensTests.cs]: merge these attributes.

File: src/Workspaces/CoreTest/CodeCleanup/AddMissingTokensTests.cs [hunk 2]
```diff
@@ -2595,6 +2595,43 @@ End Function
...
+
+    Sub Second()
+    End Sub
+End Module
+";
+
+            await VerifyAsync(code, expected);
+        }
+
+        [Fact]
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CoreTest/CodeCleanup/AddMissingTokensTests.cs]: merge Fact and Trait.

**Comment by CyrusNajmabadi** [comment]
👍

**Comment by jasonmalinowski** [comment]
This bug is still marked in the RTM milestone, why is this targeting post-dev15?

**Comment by VSadov** [comment]
@jasonmalinowski - I thought we are virtually done with RTM fixes. Do you think this will meet the bar?

**Comment by CyrusNajmabadi** [comment]
Typgin something simple in VB and you crash?  I think this could meet the bar.

**Comment by jasonmalinowski** [comment]
@VSadov The bug itself is marked as in RTM. Either the bug or the pull request need an update. I would raise the conversation at least. Although perhaps not the biggest bug, it does look pretty icky...

**Comment by VSadov** [comment]
@jasonmalinowski - will re-target the PR.

**Comment by VSadov** [comment]
@dotnet-bot test this please
I have changed the base branch.

File: src/Workspaces/CoreTest/CodeCleanup/AddMissingTokensTests.cs [hunk 3]
```diff
@@ -2594,9 +2519,45 @@ End Function
             await VerifyAsync(code, expected);
         }
 
-        [Fact]
+        [WorkItem(16781, "https://github.com/dotnet/roslyn/issues/16781")]
+        [Fact, Trait(Traits.Feature, Traits.Features.AddMissingTokens)]
```
Comment (VSadov) [inline src/Workspaces/CoreTest/CodeCleanup/AddMissingTokensTests.cs]: Fixed

**Comment by VSadov** [comment]
@dotnet-bot test this please

**Comment by VSadov** [comment]
Apparently `"()"` in PR title can cause CI to fail. Had to change title and reset tests.

**Comment by jasonmalinowski** [comment]
Why are we not fixing this in the semantic model? Have we tested other features to verify they also won't choke with the bad return?

**Comment by VSadov** [comment]
Semantic model fix is tracked by https://github.com/dotnet/roslyn/issues/16697

Here, we should not even get as far as to ask semantic model. 
"()" simply does not fit the syntax model.

**Comment by CyrusNajmabadi** [comment]
Can you comment hte code to explain as much.  Otherwise it's quite confusing.  The user then has to ask "why would i be in a location that can't support a method call, but which the semantic model says binds to a method?"  Thanks!

**Comment by VSadov** [comment]
@CyrusNajmabadi - added a comment.

**Comment by VSadov** [comment]
@dotnet-bot test this please
Reverted back to be based against post-dev15

**Comment by VSadov** [comment]
@CyrusNajmabadi @jasonmalinowski - all ok now?

**Comment by jasonmalinowski** [comment]
If this is targeting post-dev15, I'd say we should fix the semantic model.

**Comment by VSadov** [comment]
@jasonmalinowski -  There is a separate bug tracking fixing semantical model. #16697
This fix is still needed, if not for correctness, then for performance.

**Comment by jasonmalinowski** [comment]
I would be curious what measurable performance gains we see, given this isn't the only feature that touches the semantic model in this way.

More generally @CyrusNajmabadi and I found that entire function very fishy that it's opting out certain structures rather than opting in the patterns it does care about. If we're going to touch it let's just fix it for good than playing whack-a-mole for each new feature we have to add.

**Comment by VSadov** [comment]
Should I resolve the bug as WontFix?

**Comment by jaredpar** [comment]
I'm confused now.  This bug was so important that @dotnet/roslyn-ide was pushing us to take to RTW and now we're getting the message to close the PR?

**Comment by jasonmalinowski** [comment]
@jaredpar: the fix as proposed was a good tactical fix to take for RTW because the user experience would have been poor. If we're not going to take it for RTW then we should take a few extra minutes and fix up the debt here.

**Comment by CyrusNajmabadi** [comment]
> Should I resolve the bug as WontFix?

No.  The bug is very real.  The question is "what do we do about it".  In general, we want to be creating the right fixes for things.  In this case, that sounds like fixing the semantic model issue.  *However*, for RTW we were willing to take a spot-fix just to address the crash, with the understanding that the right fix would be made later.  However, since we're not going to take the spot-fix, we shoudl just do the right fix in the next iteration.

**Pull request closed without merging**